### PR TITLE
fix: Wait for more blocks before confirming anchor

### PR DESCRIPTION
**Pull request by oed**

### CODE REVIEW

**Comment by oed** [comment]
We are adding a fix in CAS instead.

**Comment by stbrody** [comment]
> We are adding a fix in CAS instead.

What does this mean?

We still occasionally see the blockTimestamp mismatch error, this was what we planned on doing to fix that.  Once we get the tests stable, it seems like we'll want to re-open this and merge it in

File: src/services/blockchain/ethereum/ethereum-blockchain-service.ts [hunk 1]
```diff
@@ -181,7 +182,7 @@ export default class EthereumBlockchainService implements BlockchainService {
     }
 
     const txReceipt: providers.TransactionReceipt = await this.wallet.provider.waitForTransaction(
-      txResponse.hash, 1, transactionTimeoutSecs * 1000);
+      txResponse.hash, NUM_BLOCKS_TO_WAIT, transactionTimeoutSecs * 1000);
```
Comment (stbrody) [inline src/services/blockchain/ethereum/ethereum-blockchain-service.ts:185]: Will we need to increase the `transactionTimeoutSecs` proportionally?  Depends on how the ethers provider works and how it applies that timeout. Does the timeout just apply for waiting for it to be initially mined?  Or does it apply to the entire `waitForTransaction` operation including waiting for extra block confirmations?

Comment (oed) [reply src/services/blockchain/ethereum/ethereum-blockchain-service.ts:185]: https://docs.ethers.io/v5/single-page/#/v5/api/providers/provider/-%23-Provider-waitForTransaction

Timeout is the frequency of queries. Confirmations is how many blocks need to be mined before the function returns.

https://github.com/ethers-io/ethers.js/blob/master/packages/providers/src.ts/base-provider.ts#L927

Comment (stbrody) [reply src/services/blockchain/ethereum/ethereum-blockchain-service.ts:185]: > Timeout is the frequency of queries. 

I don't really know what that means.  But if I'm reading the code from ethers right it looks like the timeout applies to the entire `waitForTransaction` call, including waiting for additional block confirmations.  That means we probably need to increase the timeout we're setting here proportional to the number of additional blocks we're waiting on.

Comment (stbrody) [reply src/services/blockchain/ethereum/ethereum-blockchain-service.ts:185]: to clarify, this probably isn't a code change, but rather a change in the environment variable we set for `transactionTimeoutSecs`.  Which may in turn have ripple effects on how frequently we can anchor, and on the timeouts we set in the integration tests

Comment (oed) [reply src/services/blockchain/ethereum/ethereum-blockchain-service.ts:185]: Oh i though `Timeout` specified poll interval.

Comment (stbrody) [reply src/services/blockchain/ethereum/ethereum-blockchain-service.ts:185]: Naw, that's [provider.pollingInterval](https://github.com/ceramicnetwork/ceramic-anchor-service/blob/cceb3e5cbf863b472d9fae4c33bca18239fd6ebd/src/services/blockchain/ethereum/ethereum-blockchain-service.ts#L44)

**Review by stbrody — APPROVED** [review]
LGTM, I think probably the only change we should make now is changing the `ETH_TXN_TIMEOUT` env var from 180 (3 minutes) to 240 (4 minutes), adding 1 extra minute for 4 block confirmations (one every 15 seconds). @valmack 

I pulled https://github.com/ceramicnetwork/ceramic-anchor-service/issues/360 into next sprint and added an idea to clean up and separate waiting for mining from waiting for block confirmations, which I think will help improve the stability and reliability of the CAS.  But I think for now we can just do the small change to handle the block reorg issue and that'll probably be enough to get our tests green for now.

**Comment by oed** [comment]
@stbrody do you want me to make that update in this PR?

**Comment by stbrody** [comment]
> @stbrody do you want me to make that update in this PR?

I think it makes sense to get this in quickly and do #360 shortly thereafter

**Pull request merged**